src/python/pants/backend/python/subsystems/pytest.py

```diff
@@ -19,29 +18,24 @@ def register_options(cls, register):
       help="Arguments to pass directly to Pytest, e.g. `--pytest-args=\"-k test_foo --quiet\"`",
     )
     register(
-      '--version', default='pytest>=4.6.6,<4.7', fingerprint=True,
+      '--version', default='pytest', fingerprint=True,
       help="Requirement string for Pytest.",
     )
     register(
       '--pytest-plugins',
       type=list,
       fingerprint=True,
-      default=[
-        'pytest-timeout>=1.3.3,<1.4',
-        'pytest-cov>=2.8.1,<3',
-        "unittest2>=1.1.0 ; python_version<'3'",
-        "more-itertools<6.0.0 ; python_version<'3'",
-      ],
+      default=['pytest-timeout', 'pytest-cov'],
```

Comment:
I changed these to no longer be pinned so that we default to users having the most up-to-date version of these reqs by default. If they want the stability of a pin, they can do easily do that via `--pytest-version` and `--pytest-pytest-plugins`. 

This also reduces our maintenance burden - we don't have to worry about tracking Pytest releases and keeping these versions in sync.